libs/common/comparator.hpp

```diff
@@ -0,0 +1,21 @@
+//
+// Created by Takumi Yamashita on 2017/07/13.
+//
+
+#ifndef IROHA_COMPARATOR_HPP
+#define IROHA_COMPARATOR_HPP
+
+#include <block.pb.h>
+
+namespace comparator {
+  using Block = iroha::protocol::Block;
+  class BlockComparator : public std::binary_function<Block, Block, bool> {
```

Comment:
Maybe *BlockHeightComparator* is better. The reason is because if we use a different consensus algorithm, like proof of work, then comparing block heights is not a valid block comparison (instead you would use block scores). By making the type of comparison explicit in the title, it lowers the risk of mistakes.